cmd/plakar/subcommands/mount/mount_test.go

```diff
@@ -0,0 +1,164 @@
+	require.NoError(t, err)
+	err = os.WriteFile(tmpBackupDir+"/subdir/dummy.txt", []byte("hello dummy"), 0644)
+	require.NoError(t, err)
+	err = os.WriteFile(tmpBackupDir+"/subdir/foo.txt", []byte("hello foo"), 0644)
+	require.NoError(t, err)
+	err = os.WriteFile(tmpBackupDir+"/subdir/to_exclude", []byte("*/subdir/to_exclude\n"), 0644)
+	require.NoError(t, err)
+	err = os.WriteFile(tmpBackupDir+"/another_subdir/bar", []byte("hello bar"), 0644)
+	require.NoError(t, err)
+
+	// create a storage
+	r, err := bfs.NewStore(map[string]string{"location": "fs://" + tmpRepoDir})
+	require.NotNil(t, r)
+	require.NoError(t, err)
+	config := storage.NewConfiguration()
+	serialized, err := config.ToBytes()
+	require.NoError(t, err)
+
+	hasher := hashing.GetHasher(hashing.DEFAULT_HASHING_ALGORITHM)
+	wrappedConfigRd, err := storage.Serialize(hasher, resources.RT_CONFIG, versioning.GetCurrentVersion(resources.RT_CONFIG), bytes.NewReader(serialized))
+	require.NoError(t, err)
+
+	wrappedConfig, err := io.ReadAll(wrappedConfigRd)
+	require.NoError(t, err)
+
+	err = r.Create(wrappedConfig)
+	require.NoError(t, err)
+
+	// open the storage to load the configuration
+	r, serializedConfig, err := storage.Open(map[string]string{"location": tmpRepoDir})
+	require.NoError(t, err)
+
+	// create a repository
+	ctx := appcontext.NewAppContext()
+	ctx.Stdout = bufOut
+	ctx.Stderr = bufErr
+	cache := caching.NewManager(tmpCacheDir)
+	ctx.SetCache(cache)
+
+	// Create a new logger
+	logger := logging.NewLogger(bufOut, bufErr)
+	logger.EnableInfo()
+	ctx.SetLogger(logger)
+	repo, err := repository.New(ctx, r, serializedConfig)
+	require.NoError(t, err, "creating repository")
+
+	// create a snapshot
+	snap, err := snapshot.New(repo)
+	require.NoError(t, err)
+	require.NotNil(t, snap)
+
+	imp, err := fs.NewFSImporter(map[string]string{"location": tmpBackupDir})
+	require.NoError(t, err)
+	snap.Backup(imp, &snapshot.BackupOptions{Name: "test_backup", MaxConcurrency: 1})
+
+	err = snap.Repository().RebuildState()
+	require.NoError(t, err)
+
+	return snap
+}
+
+func TestExecuteCmdMountDefault(t *testing.T) {
+	bufOut := bytes.NewBuffer(nil)
+	bufErr := bytes.NewBuffer(nil)
+
+	snap := generateSnapshot(t, bufOut, bufErr)
+	defer snap.Close()
+
+	ctx := snap.AppContext()
+	ctx.MaxConcurrency = 1
+
+	tmpMountPoint, err := os.MkdirTemp("", "tmp_mount_point")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpMountPoint)
+
+	repo := snap.Repository()
+	// override the homedir to avoid having test overwriting existing home configuration
+	ctx.HomeDir = repo.Location()
+	args := []string{tmpMountPoint}
+
+	subcommand, err := parse_cmd_mount(ctx, repo, args)
+	require.NoError(t, err)
+	require.NotNil(t, subcommand)
+	require.Equal(t, "mount", subcommand.(*Mount).Name())
+
+	go func() {
+		status, err := subcommand.Execute(ctx, repo)
+		require.NoError(t, err)
+		require.Equal(t, 0, status)
+	}()
+
+	// wait for the goroutine to be executed
+	time.Sleep(1 * time.Second)
```

Comment:
can't it wait on a channel to be closed by the goroutine instead ?

Reply:
@poolpOrg the issue is that the goroutine will never finish, as once executed the mount command awaits until we Ctrl+C . So using a channel or a waitgroup will make the test hang indefinitely